### PR TITLE
SAAS-3071: Added MQ Broker Encrypted Plugin

### PR DESCRIPTION
**Pull request by sadeed12345**

### CODE REVIEW

File: plugins/aws/mq/mqBrokerEncrypted.js [hunk 1]
```diff
@@ -0,0 +1,125 @@
...
+        };
+
+        var desiredEncryptionLevel = helpers.ENCRYPTION_LEVELS.indexOf(config.desiredEncryptionLevelString);
+        var currentEncryptionLevel;
+
+        async.each(regions.mq, function(region, rcb){        
+            var listBrokers = helpers.addSource(cache, source,
+                ['mq', 'listBrokers', region]);
+                    
+
```
Comment (sadeed12345) [inline plugins/aws/mq/mqBrokerEncrypted.js:39]:
```suggestion
```

File: plugins/aws/mq/mqBrokerEncrypted.js [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+                    var describeKey = helpers.addSource(cache, source,
+                        ['kms', 'describeKey', region, keyId]);  
+
+                    if (!describeKey || describeKey.err || !describeKey.data || !describeKey.data.KeyMetadata) {
+                        helpers.addResult(results, 3,
+                            `Unable to query KMS key: ${helpers.addError(describeKey)}`,
+                            region, KmsKeyId);
+                        continue;
+
```
Comment (sadeed12345) [inline plugins/aws/mq/mqBrokerEncrypted.js:95]:
```suggestion
```

File: plugins/aws/mq/mqBrokerEncrypted.js [hunk 3]
```diff
@@ -0,0 +1,124 @@
...
+                        which is greater than or equal to the desired encryption level ${config.desiredEncryptionLevelString}`,
+                        region, resource);
+                } else {
+                    helpers.addResult(results, 2,
+                        `MQ Broker Data at-rest is encrypted with ${currentEncryptionLevelString} \
+                        which is less than the desired encryption level ${config.desiredEncryptionLevelString}`,
+                        region, resource);
+                }
+            }
+
```
Comment (sadeed12345) [inline plugins/aws/mq/mqBrokerEncrypted.js:117]:
```suggestion
```

File: plugins/aws/mq/mqBrokerEncrypted.spec.js [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+            mqBrokerEncrypted.run(cache, { mq_broker_desired_encryption_level : 'awscmk' }, (err, results) => {
+                expect(results.length).to.equal(1);
+                expect(results[0].status).to.equal(0);
+                expect(results[0].region).to.equal('us-east-1');
+                done();
+            });
+        });
+
+        it('should FAIL if MQ Broker data at-rest is not encrypted with desired encryption level', function (done) {
+            const cache = createCache([listBrokers[1]],listKeys, [describeBroker[0]], describeKey[1]);
```
Comment (dipsubha06) [inline plugins/aws/mq/mqBrokerEncrypted.spec.js:190]: describeBroker[0] has a different key which is not there in describeKey or listKeys. is this test working as expected if so how?

**Review by dipsubha06 — CHANGES_REQUESTED** [review]

File: plugins/aws/mq/mqBrokerEncrypted.js [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+    category: 'MQ',
+    domain: 'Application Integration',
+    description: 'Ensure that Amazon MQ brokers have data ecrypted at-rest feature enabled.',
+    more_info: 'Amazon MQ encryption at rest provides enhanced security by encrypting your data using encryption keys stored in the AWS Key Management Service (KMS).',
+    recommended_action: 'Enabled data at-rest encryption feature for MQ brokers',
+    link: 'https://docs.aws.amazon.com/amazon-mq/latest/developer-guide/data-protection.html#data-protection-encryption-at-rest',
+    apis: ['MQ:listBrokers', 'MQ:describeBroker', 'KMS:describeKey', 'KMS:listKeys'],
+    settings: {
+        mq_broker_desired_encryption_level: {
+            name: 'MQ Broker Encrypted',
```
Comment (AkhtarAmir) [inline plugins/aws/mq/mqBrokerEncrypted.js:15]:
```suggestion
            name: 'MQ Broker Target Encryption Level',
```

File: plugins/aws/mq/mqBrokerEncrypted.js [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+            description: 'In order (lowest to highest) awskms=AWS-managed KMS; awscmk=Customer managed KMS; externalcmk=Customer managed externally sourced KMS; cloudhsm=Customer managed CloudHSM sourced KMS',
+            regex: '^(awskms|awscmk|externalcmk|cloudhsm)$',
+            default: 'awscmk'
+        }
+    },
+
```
Comment (AkhtarAmir) [inline plugins/aws/mq/mqBrokerEncrypted.js:21]:
```suggestion
```

**Review by dipsubha06 — APPROVED** [review]

**Review by dipsubha06 — APPROVED** [review]

**Pull request merged**